adapters/thetradedesk/thetradedesk.go

```diff
@@ -0,0 +1,184 @@
+package thetradedesk
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+)
+
+type adapter struct {
+	URI        string
+	bidderName string
+}
+
+type ExtImpBidderTheTradeDesk struct {
+	adapters.ExtImpBidder
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	pubID, err := getPublisherId(request.Imp)
+
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	for i := 0; i < len(request.Imp); i++ {
+		imp := request.Imp[i]
+
+		if imp.Banner != nil {
+			if len(imp.Banner.Format) > 0 {
+				firstFormat := imp.Banner.Format[0]
+				imp.Banner.H = &firstFormat.H
+				imp.Banner.W = &firstFormat.W
+
+			}
+		}
+	}
+
+	if request.Site != nil {
+		siteCopy := *request.Site
+		if siteCopy.Publisher != nil {
+			publisherCopy := *siteCopy.Publisher
+			publisherCopy.ID = pubID
+			siteCopy.Publisher = &publisherCopy
+		} else {
+			siteCopy.Publisher = &openrtb2.Publisher{ID: pubID}
+		}
+		request.Site = &siteCopy
+	} else if request.App != nil {
+		appCopy := *request.App
+		if appCopy.Publisher != nil {
+			publisherCopy := *appCopy.Publisher
+			publisherCopy.ID = pubID
+			appCopy.Publisher = &publisherCopy
+		} else {
+			appCopy.Publisher = &openrtb2.Publisher{ID: pubID}
+		}
+		request.App = &appCopy
+	}
+
+	var bidderEndpoint = strings.TrimRight(a.URI, "/") + "/" + a.bidderName
+
+	errs := make([]error, 0, len(request.Imp))
+	reqJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	PREBID_INTEGRATION_TYPE := "1"
```

Comment:
This is a constant. Should be declared top of this file as - 
const PREBID_INTEGRATION_TYPE = "1"

Reply:
moved tot he top.